libs/unity/library/Runtime/Scripts/Media/MicrophoneSource.cs

```diff
@@ -75,6 +107,47 @@ protected async void OnEnable()
             AudioSourceStarted.Invoke(this);
         }
 
+#if PLATFORM_ANDROID
+        protected void OnApplicationFocus(bool hasFocus)
+        {
+            if (!hasFocus)
+            {
+                return;
+            }
+
+            // If focus is restored after a pending request, check the permission again
+            if (_androidRecordAudioRequestPending)
+            {
+                _androidRecordAudioRequestPending = false;
+
+                if (Permission.HasUserAuthorizedPermission(Permission.Microphone))
+                {
+                    // If now authorized, start capture as if just enabled
+                    Debug.Log("User granted authorization to access microphone, starting MicrophoneSource now...");
+                    OnEnable();
```

Comment:
nit: extract video initialization in its own function rather than calling OnEnable again - shouldn't make much difference but flow looks clearer IMHO.

Reply:
I did it like this on purpose to make sure no one introduces some code in `OnEnable()` that would not be run through this workaround. I agree with you usually I would have extracted a separate function, but here I want to avoid future troubles.